test/tests/functional/pbs_release_limited_res_suspend.py

```diff
@@ -0,0 +1,352 @@
+        except PbsSignalError, e:
+            self.assertEqual(rc, 0, e.msg[0])
+
+        ras_mem = ATTR_rescassn + '.mem'
+        ras_ncpus = ATTR_rescassn + '.ncpus'
+
+        rv = self.server.status(
+            NODE, [ras_ncpus, ras_mem], id=self.mom.shortname)
+        self.assertNotEqual(rv, None)
+
+        self.assertEqual(rv[0][ras_mem], "524288kb",
+                         msg="pbs should not release memory")
+        self.assertEqual(rv[0][ras_ncpus], "0",
+                         msg="pbs did not release ncpus")
+
+    def test_change_in_res_to_release_on_suspend(self):
+        """
+        set restrict_res_to_release_on_suspend to only ncpus and then suspend
+        a job after the job is suspended change
+        restrict_res_to_release_on_suspend to release only memory and check
+        if the suspended job resumes and do not account for memory twice.
+        """
+
+        # Set restrict_res_to_release_on_suspend server attribute
+        a = {'restrict_res_to_release_on_suspend': 'ncpus'}
+        self.server.manager(MGR_CMD_SET, SERVER, a, expect=True)
+
+        # Submit a low priority job
+        j1 = Job(TEST_USER)
+        j1.set_attributes({'Resource_List.select': '1:ncpus=4:mem=512mb'})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+
+        # Submit a high priority job
+        j2 = Job(TEST_USER)
+        j2.set_attributes(
+            {'Resource_List.select': '1:ncpus=2:mem=512mb',
+             ATTR_q: 'expressq'})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'S', 'substate': 45}, id=jid1)
+
+        # Change restrict_res_to_release_on_suspend server attribute
+        a = {'restrict_res_to_release_on_suspend': 'mem'}
+        self.server.manager(MGR_CMD_SET, SERVER, a, expect=True)
+
+        rc = 0
+        try:
+            rc = self.server.deljob(jid2, wait=True)
+        except PbsDeljobError, e:
+            self.assertEqual(rc, 0, e.msg[0])
+
+        self.server.expect(JOB, {'job_state': 'R', 'substate': 42}, id=jid1)
+        ras_mem = ATTR_rescassn + '.mem'
+        ras_ncpus = ATTR_rescassn + '.ncpus'
+
+        rv = self.server.status(
+            NODE, [ras_ncpus, ras_mem], id=self.mom.shortname)
+        self.assertNotEqual(rv, None)
+
+        self.assertEqual(rv[0][ras_mem], "524288kb",
+                         msg="pbs did not account for memory correctly")
+        self.assertEqual(rv[0][ras_ncpus], "4",
+                         msg="pbs did not account for ncpus correctly")
+
+    def test_res_released_sched_susp(self):
+        """
+        Test if job's resources_released attribute is correctly set when
+        it is suspended.
+        """
+
+        # Set restrict_res_to_release_on_suspend server attribute
+        a = {'restrict_res_to_release_on_suspend': 'ncpus'}
```

Comment:
You use the ATTR_* variables further down, but call this attribute out by name.  I'm not sure I saw you add an ATTR_* PTL keyword for it in the code.  Did you?  If you did, why not use it here?